linera-core/src/client/mod.rs

```diff
@@ -675,21 +677,23 @@ where
     /// local chain.
     #[tracing::instrument(level = "trace")]
     async fn pending_message_bundles(&self) -> Result<Vec<IncomingBundle>, ChainClientError> {
-        if self.next_block_height() != BlockHeight::ZERO && self.options.message_policy.is_ignore()
-        {
-            return Ok(Vec::new()); // OpenChain is already received, other are ignored.
-        }
```

Comment:
Why did you move it, was it wrong?

Reply:
Our `ChainState`'s `next_block_height` could be out of sync with the `local_node`. We should use the latest information from the local node to make decisions.

Reply:
The only part that matters here is whether `next_block_height` is non-ZERO - in which case we should return `Ok(Vec::new())` – this is only the case when a chain makes THE FIRST the new block before last update of `self.next_block_height` and the query. Does that sound correct?

Reply:
Yes.
So here it could happen that the chain state is stale and thinks we're still at block height 0, but the chain info knows we're already at block height 1. In that case it makes sense to return `Vec::new()`.

Reply:
We update `self.next_block_height` in a couple of places so it's not "stale forever".